portfolio/src/main/webapp/script.js

```diff
@@ -44,3 +44,14 @@ function addRandomFact() {
   const factContainer = document.getElementById('fact-container');
   factContainer.innerText = fact;
 }
+
+/**
+ * Fetches greeting from /data for use as opening greeting
+ */
+async function getOpeningGreeting() {
+  const response = await fetch('/data');
+  const greeting = await response.text();
+  let innerHTML = document.getElementById('opening-greeting').innerHTML;
```

Comment:
curious - why does the existing innerHtml need to be preserved? This will be empty on page load, so isn't this effectively a no-op?

Reply:
Oops that was a lapse in reasoning, I was initially manipulating the body element directly but changed the code to manipulate the container instead. So yes this is a no-op. I'll remove it.